raftinc/basicparallel.hpp

```diff
@@ -24,49 +24,203 @@
 #include "kernel.hpp"
 #include "port_info.hpp"
 #include "fifo.hpp"
-#include "allocate.hpp"
 #include "schedule.hpp"
+#include "mapbase.hpp"
+#include "allocate.hpp"
+#include "common.hpp"
+#include "streamingstat.tcc"
+#include "graphtools.hpp"
 #include <set>
+#include <map>
 #include <iostream>
 
-namespace raft
-{
-    class map;
-}
 
 /** right now we're only considering single input, single output kernels **/
 struct stats
 {
-   std::uint16_t occ_in  = 0;
-   float        service_rate = static_cast< float >( 0 );
-   friend   std::ostream& operator << ( std::ostream &stream, stats &s )
-   {
-      stream << "occupancy count: " << s.occ_in << "\n";
-      stream << "service rate: " << s.service_rate << "\n";
-      return( stream );
-   }
+    std::uint16_t occ_in  = 0;
```

Comment:
@jonathan-beard , double check to see if the service rate instrumentation and such is still functional. I think @BamboWu was primarily using the enqueue/dequeue blocked stats, would be nice to have the service rate estimates going again for initial transport buffer sizing.